### PR TITLE
New Plugin for dotnet CLI

### PR DESCRIPTION
**Pull request by clebron949**
This PR is copying the dotnet plugin from oh-my-zsh. This plugin is handy for dotnet developers.  I used it so much in oh-my-zsh that I had to add it here.

### CODE REVIEW

File: plugins/dotnet/dotnet.plugin.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# This scripts is copied from (MIT License):
```
Comment (akinomyoga) [inline plugins/dotnet/dotnet.plugin.sh:1]: We add the following header line to every plugin file.

```suggestion
#! bash oh-my-bash.module
#
# This scripts is copied from (MIT License):
```

Also, there was a link after this line in the upstream OMZ, but the link is dropped in this PR. Then, there is nothing after "from". Please remove this line or add a link to complete the sentence.

**Review by akinomyoga — COMMENTED** [review]
Thank you for your contribution. I need to check the change history of the upstream to check the copyright holders. The MIT license requires retaining the original copyright notice of the form "Copyright (c) \<year> \<name>", but it seems non-trivial whether this PR properly does that. The mention of OMZ's copyright is already included in our `LICENSE.md`, but the upstream also mentions the link in `dotnet/sdk`. The codes from `dotnet/sdk` seems related to the completion setting, which was dropped in this PR. On the other hand, the upstream OMZ appears to contain the name of the original author of the aliases, which was also dropped in this PR.

File: plugins/dotnet/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# .NET Core CLI plugin
+
+This plugin provides completion and useful aliases for [.NET Core CLI](https://dotnet.microsoft.com/).
```
[the remark this answers is not in the document]
Comment (clebron949) [reply plugins/dotnet/README.md:3]: Thank you for your feedback. I added the completion in commit [9fc7939](https://github.com/ohmybash/oh-my-bash/pull/597/commits/9fc79396ff44502ad679d844e06e40025dbe4b55).

[the remark this answers is not in the document]
Comment (clebron949) [reply plugins/dotnet/README.md]: I removed the completions for now. It failed the test I created for them. Once I'll get it working, I'll create a new request.

[the remark this answers is not in the document]
Comment (akinomyoga) [reply plugins/dotnet/README.md]: .NET SDK provides also a Bash version at

https://raw.githubusercontent.com/dotnet/sdk/main/scripts/register-completions.bash

[the remark this answers is not in the document]
Comment (akinomyoga) [reply plugins/dotnet/README.md:3]: Thank you!

File: plugins/dotnet/dotnet.plugin.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#! bash oh-my-bash.module
+#
+# This plugin is based on the following version of the Oh-My-Zsh plugin:
+# https://github.com/ohmyzsh/ohmyzsh/blame/c68ff8aeedc2b779ae42d745457ecd443e22e212/plugins/dotnet/dotnet.plugin.zsh
+# bash parameter completion for the dotnet CLI was taken from the following
```
Comment (akinomyoga) [inline plugins/dotnet/dotnet.plugin.sh:5]:
```suggestion
# https://github.com/ohmyzsh/ohmyzsh/blame/c68ff8aeedc2b779ae42d745457ecd443e22e212/plugins/dotnet/dotnet.plugin.zsh

# bash parameter completion for the dotnet CLI was taken from the following
```

Sorry, could you add an empty line here? The handling of an empty line by the GitHub suggestion UI seems to be broken...

**Comment by akinomyoga** [comment]
Thank you. I've merged it.

**Pull request merged**